### PR TITLE
Improve typography accessibility

### PR DESCRIPTION
**Pull request by prateek3255**
- ✨ Update the Text API to accept `as` prop that can render any HTML element
- ♻️ Update already existing Text components to use the newer API taking care of the semantic HTML tags being used.

### CODE REVIEW

File: theme/theme.ts [hunk 1]
```diff
@@ -7,12 +7,15 @@ export type Theme = 'blue' | 'teal' | 'green' | 'indigo' | 'orange' | 'purple' |
...
-};
+  '3xl': `font-sniglet text-6xl font-extrabold`,
+  '2.5xl': `font-sniglet text-5xl font-extrabold`,
+  '2xl': `font-sniglet text-4xl font-extrabold`,
+  '1.5xl': `font-sniglet text-3xl font-extrabold`,
+  xl: `font-sniglet text-2xl font-extrabold`,
+  lg: `font-sniglet text-xl`,
+  md: `font-roboto text-base`,
+  sm: `font-roboto text-sm`,
+} as const;
```
Comment (gupta-ji6) [inline theme/theme.ts:18]:
```suggestion
  6xl: `font-sniglet text-6xl font-extrabold`,
  5xl: `font-sniglet text-5xl font-extrabold`,
  4xl: `font-sniglet text-4xl font-extrabold`,
  3xl: `font-sniglet text-3xl font-extrabold`,
  2xl: `font-sniglet text-2xl font-extrabold`,
  xl: `font-sniglet text-xl`,
  base: `font-roboto text-base`,
  sm: `font-roboto text-sm`,
} as const;
```
@prateek3255 I'm a little uncertain about this. The key names should ideally match the tailwind's text font size class name.

why did we decide to have 1.5xl? 🤔

File: components/common/Text.tsx [hunk 2]
```diff
@@ -2,22 +2,22 @@ import { TextType } from '../../theme/theme';
...
   additionalStyles?: string;
-  inline?: boolean;
 }): JSX.Element => {
-  const textType = TextType[type];
+  const textType = TextType[size];
   const styles = `${textType} ${color} ${additionalStyles}`;
-
-  return inline ? <span className={styles}>{children}</span> : <div className={styles}>{children}</div>;
+  const Tag = as;
+  return <Tag className={styles}>{children}</Tag>;
```
Comment (gupta-ji6) [inline components/common/Text.tsx:20]: this looks like a nice API now which is accessible too - amazing 👌🏻

File: theme/theme.ts (hunk 1 above)
Comment (prateek3255) [reply theme/theme.ts:18]: Yeah, the initial naming was like that, and when I started refactoring, I went along with the initial naming and ended up with this 😓. I'll fix this ✅

File: components/common/Text.tsx (hunk 2 above)
Comment (prateek3255) [reply components/common/Text.tsx:20]: 🥳

File: theme/theme.ts (hunk 1 above)
Comment (prateek3255) [reply theme/theme.ts]: Fixed 👍🏼

**Review by gupta-ji6 — DISMISSED** [review]
LGTM now, let's merge & deploy 🚀

**Comment by gupta-ji6** [comment]
@prateek3255 need to bump the version in `package.json` before merging.

**Comment by gupta-ji6** [comment]
@prateek3255 we should think upon automating version changes via https://github.com/changesets/changesets or something 🤔 
what's say?

there might be a GH Action as well for the same.

**Review by gupta-ji6 — APPROVED** [review]

**Comment by prateek3255** [comment]
> @prateek3255 we should think upon automating version changes via https://github.com/changesets/changesets or something 🤔 what's say?
> 
> there might be a GH Action as well for the same.

Yes I have actually seen GitHub actions/bots for this that automatically bump version write release notes and publish the library, we can create one on our own as well

**Pull request merged**